src/main/groovy/nextflow/processor/TaskProcessor.groovy

```diff
@@ -1793,6 +1794,19 @@ class TaskProcessor {
                     contextMap.put( param.name, val )
                     break
 
+                case FilemapInParam:
+                    def normalized = normalizeInputToFiles(val.values(),count)
+                    def mapfile = val
+                    int i = 0
+                    mapfile.keySet().each {
+                        mapfile[it] = Paths.get(normalized[i++].stageName)
+                    }
+
+                    contextMap.put( param.name,  mapfile)
+                    count += normalized.size()
+                    task.setInput(param, normalized)
+                    return
+
```

Comment:
is [this](https://github.com/nextflow-io/nextflow/blob/master/src/main/groovy/nextflow/processor/TaskProcessor.groovy#L1594-1594) method doing that. What I'm not understanding is how you are handling metadata with this path. Can you comment here #343 ?